### PR TITLE
feat(exhaust.ts): renamed exhaust to exhaustAll

### LINKED ISSUE #4573: [docs]: Rename/Alias the exhaust operator to exhaustAll
## Bug Report

**Current Behavior**
All good! Just the naming might be confusing.

**Possible Solution**
I would expect the name of exhaust to be exhaustAll.

**Additional context/Screenshots**
If you take a look into the code you will realize that all the other operators like switchAll, mergeAll, zipAll etc. process higher order observables like exhaust do. 

Therefore I consider the naming as inconsistent. 

I would suggest to implement it as an alias.

### PR DESCRIPTION
**Pull request by BioPhoton**
**Description:**

As proposed in #4573 I renamed `exhaust` to `exhaustAll` and exposed the old as an alias that I deprecated.

Notice: 
This PR changes the public API and therefore I had to run `npm run api_guardian:update`. 
I added the generated changes in a separate commit.

Things changed:
- renamed `exhaust` to `exhaustAll` 
- exported `exhaustAll` with alias `exhaust`
- tested export of `exhaustAll` and `exhaust`
- deprecated `exhaust`
- updated tests to use new method name
- updated `api_guard/dist/types/operators/index.d.ts` by running `npm run api_guardian:update`

**Related issue (if exists):**
closes #4573

### CODE REVIEW

File: src/operators/index.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export { distinctUntilKeyChanged } from '../internal/operators/distinctUntilKeyC
 export { elementAt } from '../internal/operators/elementAt';
 export { endWith } from '../internal/operators/endWith';
 export { every } from '../internal/operators/every';
-export { exhaust } from '../internal/operators/exhaust';
+export { exhaust, exhaustAll } from '../internal/operators/exhaust';
```
Comment (jakovljevic-mladen) [inline src/operators/index.ts:31]: Hi. Does it make sense to rename `exhaust.ts` file to `exhaustAll.ts`? Would such change break anything?

Comment (BioPhoton) [reply src/operators/index.ts:31]: I believe not as it lives in the internal folder. Good point!
After figuring out how to fix the breaking build ill ask for it.

**Comment by jakovljevic-mladen** [comment]
Cool, thanks @BioPhoton. I'm sorry, but there's another file left that could be renamed:  `spec/operators/exhaust-spec.ts` ->  `spec/operators/exhaustAll-spec.ts`.

**Comment by benlesh** [comment]
Unfortunately, this all may need redone after #5729 lands. Marking as blocked for now.

**Comment by BioPhoton** [comment]
Ok. No problem at all. I'll skip the last reaming and redo everything when you give me the GO.

👍

**Comment by benlesh** [comment]
Okay, @BioPhoton, give it a go now. You can probably reuse parts of this PR, so I'll leave it open.

Thank you for your hard work. (Ping me on Twitter if I don't notice when you update this)

**Comment by cartant** [comment]
@benlesh I've rebased this. It should to okay to merge once it passes CI.

**Review by cartant — APPROVED** [review]

**Review by benlesh — APPROVED** [review]

**Comment by benlesh** [comment]
lol... finally got this in.

**Pull request merged**